compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/cfg/ControlFlowGraphBuilder.kt

```diff
@@ -804,7 +804,15 @@ class ControlFlowGraphBuilder {
         levelCounter--
         val node = createTryMainBlockExitNode(tryExpression)
         popAndAddEdge(node)
-        addEdge(node, tryExitNodes.top())
+        // NB: If this block can escape, not needed to connect to the finally block.
+        val tryMainBlockEscaped = blockEscaped(node) { n -> n is TryMainBlockEnterNode && n.level == levelCounter }
```

Comment:
I still don't agree with that this check. Why it needed?  If all exits from try main blocks are returns then execution still be passed to finally before return so there **should** be an edge from that jump to finally

Reply:
Thank you for keeping pointing out this. It seems like I still didn't fully comprehend try-catch-finally. :( Now I'm convinced, and removing this revealed that one of changes in `neg/3` was actually wrong. Also, interpretation in the analyzer that only checks the normal flow was wrong. Both places are fixed, thanks to this place. Reverted escaping checks.